src/plugins/gui/amountQuotePlugin.ts

```diff
@@ -476,7 +483,16 @@ export const amountQuoteFiatPlugin: FiatPluginFactory = async (params: FiatPlugi
           const noQuoteText = direction === 'buy' ? lstrings.fiat_plugin_buy_no_quote : lstrings.fiat_plugin_sell_no_quote
           if (goodQuotes.length === 0) {
             // Find the best error to surface
-            const bestError = getBestError(errors as any, sourceFieldCurrencyCode, direction)
+            const fpQuoteErrors = errors
+              .map(error => {
+                if (error instanceof FiatProviderError) {
+                  return error.quoteError
+                }
+                return error
```

Comment:
Won't this return a `!= null` value, so the filter below will let the wrong thing through?

Reply:
This was originally expecting `error` to be a `FiatProviderError.` In Moonpay's case, was passing a `FiatProviderQuoteError,` the child object within the former.

This change tries to maintain the original expectations should it arise for other plugins.

1. Handle `FiatProviderErrors`
2. Handle `FiatProviderQuoteErrors`
3. Fallback to a default error for other situations